src/traces.jl

```diff
@@ -133,56 +119,200 @@ end
 
 #####
 
-struct MergedTraces{names,T,N,E} <: AbstractTraces{names,E}
+"""
+    Episode(traces)
+
+An `Episode` is a wrapper around [`Traces`](@ref). You can use `(e::Episode)[]`
+to check/update whether the episode reaches a terminal or not.
+"""
+struct Episode{T,names,E} <: AbstractTraces{names,E}
+    traces::T
+    is_terminated::Ref{Bool}
+end
+
+Episode(t::AbstractTraces{names,T}) where {names,T} = Episode{typeof(t),names,T}(t, Ref(false))
+
+@forward Episode.traces Base.getindex, Base.setindex!, Base.size
+
+Base.getindex(e::Episode) = getindex(e.is_terminated)
+Base.setindex!(e::Episode, x::Bool) = setindex!(e.is_terminated, x)
+
+for f in (:push!, :append!)
+    @eval function Base.$f(t::Episode, x)
+        if t.is_terminated[]
+            throw(ArgumentError("The episode is already flagged as done!"))
+        else
+            $f(t.traces, x)
+        end
+    end
+end
+
+function Base.pop!(t::Episode)
+    pop!(t.traces)
+    t.is_terminated[] = false
+end
+
+Base.pushfirst!(t::Episode, x) = pushfirst!(t.traces, x)
+Base.prepend!(t::Episode, x) = prepend!(t.traces, x)
+Base.popfirst!(t::Episode) = popfirst!(t.traces)
+
+function Base.empty!(t::Episode)
+    empty!(t.traces)
+    t.is_terminated[] = false
+end
+
+#####
+
+"""
+    Episodes(init)
+
+A container for multiple [`Episode`](@ref)s. `init` is a parameterness function which return an empty [`Episode`](@ref).
+"""
+struct Episodes{names,E} <: AbstractTraces{names,E}
+    init::Any
+    episodes::Vector{Episode}
```

Comment:
Here we should add the parameters of Episode to avoid having a collection with non-concrete type as a field (this may lead to type instability).
```suggestion
struct Episodes{T, names,E} <: AbstractTraces{names,E}
    init::Any
    episodes::Vector{Episode{T, names, E}}
```

Reply:
Added